### PR TITLE
gsg, tutorial, release_notes, references: fixes and new contents

### PR DESCRIPTION
**Pull request by ttzeng**
This PR proposes the following changes:
1. Fix the warning while building the release notes.
2. Fix the missing reference link of ACRN DM due to re-organize the project source hierarchy.
3. Reformat the Thermal Daemon tutorial.
4. Add the Celadon patch submission guide.

Signed-off-by: Tonny Tzeng <tonny.tzeng@intel.com>
Signed-off-by: Gnanasekaran, Sundar <sundar.gnanasekaran@intel.com>
Signed-off-by: Balan, Swaroop <swaroop.balan@intel.com>
Reviewed-by: Kristi Rifenbark <kristi@buzzcollectivemarketing.com>

### CODE REVIEW

**Comment by ttzeng** [comment]
@klynnrif please help to review and polish the thermal_daemon tutorial and the patch submission guide proposed in this PR. Thanks.

File: source/getting_started/submit-patch.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+github     Good to go with github PR push process
+aosp       Create a google_diff patch & in-parallel
+           up-stream to aosp
+kernelorg  Upstream to kernel.org
+graphics   Upstream to graphics remote
+intel      Upstream to intel remote
+trusty-ia  Upstream to trusty-ia remote
+=========  ========================================
+
+You can further enter the following command to identify which branch the source head is pointing to:
```
Comment (klynnrif) [inline source/getting_started/submit-patch.rst]: Suggested rewrite: 
Further, you can enter the following commands to identify which branch the source head points to:

File: source/getting_started/submit-patch.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+.. code-block:: bash
+
+   $ git remote show <remote-alias> | grep 'HEAD branch:'
+
+|C| Branching Strategy
+----------------------
+
+There are *master branch* and *stable branch* for some of |C| repositories. Developers should make changes on the master branch, the CoE (Center of Excellence) lead of the repo will pull them from the master branch to the stable branch once it is stabilised. Please review the manifest include file *.repo/manifests/include/bsp-celadon.xml*, which tells the branch info that the project source code is branched off. For example, the source code of *art-extenstion* project and *trusty* projects are based on the stable branch and master branch respectively:
```
Comment (klynnrif) [inline source/getting_started/submit-patch.rst]: Suggested rewrite: 
There is a *master branch* and *stable branch* for some of |C| repositories. Make developer changes on the master branch. The CoE (i.e. Center of Excellence) lead of the repo pulls them from the master branch to the stable branch once it is stabilized. Please review the manifest include file *.repo/manifests/include/bsp-celadon.xml*, which tells the branch info that the project source code is branched off. For example, the source code of the *art-extenstion* project and *trusty* projects are based on the stable branch and master branch respectively:

File: source/getting_started/submit-patch.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+   <project name="vendor_intel_art-extension" path="vendor/intel/art-extension" remote="github" revision="stable"/>
+   ...
+   <project name="trusty_external_headers" path="trusty/external/headers" remote="trusty-ia" revision="master"/>
+   <project name="trusty_external_lk" path="trusty/external/lk"  remote="trusty-ia" revision="master">
+   ...
+
+Submit Patches
+--------------
+
+Once you figured out the remote repo to fetch the original source code, and which branch the software is branched off, you can set up a local workspace by `cloning the source repository <https://help.github.com/articles/fork-a-repo>`_ on your computer. If you'd like to propose changes to any non-AOSP project repository, you need to `create a Pull Request <https://help.github.com/articles/creating-a-pull-request-from-a-fork>`_, and send the URL of the PR to the domain owners for review. You may revise and `update the submitted PRs <https://help.github.com/articles/changing-a-commit-message>`_ to address the review feedbacks from the domain owners if any.
```
Comment (klynnrif) [inline source/getting_started/submit-patch.rst]: Suggested rewrite: 
Once you determine which remote repository to fetch the original source code from and which branch the software has branched off of, you can set up a local workspace. Do this by `cloning the source repository <https://help.github.com/articles/fork-a-repo>`_ on your computer. If you would like to propose changes to any non-AOSP project repository, `create a Pull Request <https://help.github.com/articles/creating-a-pull-request-from-a-fork>`, and send the URL of the PR to the domain owners for review. You may revise and `update the submitted PRs <https://help.github.com/articles/changing-a-commit-message>`_ to address any review feedback from the domain owners.

File: source/getting_started/submit-patch.rst [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+It's recommended to create commit messages in the following format while submitting your code to |C|:
```
Comment (klynnrif) [inline source/getting_started/submit-patch.rst]: Suggested rewrite: 
We recommend you create commit messages in the following format when you submit your code to |C|:

File: source/getting_started/submit-patch.rst [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+   
+   <Description>
+   
+   Tracked-On:
+   Signed-off:
+
+Submit AOSP Patches
+~~~~~~~~~~~~~~~~~~~
+
+Patches against AOSP project source code use different approach. Follow below steps to create your *google_diff* patch set if you need to make changes on any AOSP project.
```
Comment (klynnrif) [inline source/getting_started/submit-patch.rst]: Suggested rewrite: 
Patches against the AOSP project source code use a different approach. Complete the following steps to create your *google_diff* patch set to make any changes on an AOSP project.

File: source/getting_started/submit-patch.rst [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+#. Clone the AOSP project to your local workspace, make changes and create commit messages in the workspace.
```
Comment (klynnrif) [inline source/getting_started/submit-patch.rst]: Suggested rewrite: 
Clone the AOSP project to your local workspace. Make your changes and create your commit messages in the workspace.

File: source/getting_started/submit-patch.rst [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+#. Generate the patch set using the following command, where *n* is the number of commits added in the previous step.
+
+   .. code-block:: none
+
+      $ git format-patch HEAD~<n>
+
+#. Copy the patch files generated in the previous step to the *vendor/intel/utils/android_p/google_diff/<target_product>/<aosp-repo>/* folder in the |C| source tree. You should maintain the patch number in sequence, and create the above folder if it doesn't exist.
```
Comment (klynnrif) [inline source/getting_started/submit-patch.rst]: Suggested rewrite: 
Copy the patch files generated in the previous step to the *vendor/intel/utils/android_p/google_diff/<target_product>/<aosp-repo>/* folder in the |C| source tree. Maintain the patch number in sequence and create the previous folder if it does not exist.

File: source/tutorials/thermal-daemon/thermal-daemon.rst [hunk 8]
```diff
@@ -0,0 +1,66 @@
+.. _thermal-daemon:
+
+Thermal Daemon: Intel thermal solution for |C|
+==============================================
+
+Introduction
+------------
+
+`Thermal daemon (TD) <https://github.com/intel/thermal_daemon>`_ is an opensource thermal solution provided by Intel. It is a service that gets started at Android filesystem bootup and executes throughout the uptime of the system. It supports multiple thermal policies such as active, passive and critical, users can configure thermal policies for a platform using the *thermal-conf.xml* in the */vendor/etc/thermal_daemon* directory on the target.
```
Comment (klynnrif) [inline source/tutorials/thermal-daemon/thermal-daemon.rst]: Suggested rewrite: 
`Thermal daemon (TD) <https://github.com/intel/thermal_daemon>`_ is an open source thermal solution provided by Intel. It is a service that starts at the Android filesystem boot up and executes throughout the uptime of the system. It supports multiple thermal policies such as active, passive, and critical. Users can configure thermal policies for a platform using *thermal-conf.xml* in the */vendor/etc/thermal_daemon* directory on the target.

File: source/tutorials/thermal-daemon/thermal-daemon.rst [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+Thermal daemon relies on the thermal sensor temperature inputs in Linux thermal sysfs interfaces */sys/class/thermal*, and based on that, it takes appropriate throttling/cooling actions as defined in the *thermal-conf.xml*. Reference the man pages for the `Thermal daemon details <https://github.com/intel/thermal_daemon/blob/master/man/thermald.8>`_, and the `thermal-conf.xml configuration file options <https://github.com/intel/thermal_daemon/blob/master/man/thermal-conf.xml.5>`_.
+
+Kernel drivers used by Thermal Daemon solution on |C|
+-----------------------------------------------------
+
+Thermal daemon looks for thermal sensors and thermal cooling drivers in the Linux kernel. The following kernel drivers are used on Celadon:
```
Comment (klynnrif) [inline source/tutorials/thermal-daemon/thermal-daemon.rst]: Capitalize "daemon" / Thermal Daemon

File: source/tutorials/thermal-daemon/thermal-daemon.rst [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+======================================  ===============================
+drivers/thermal/x86_pkg_temp_thermal.c  SOC package temperature driver
+drivers/powercap/intel_rapl.c           RAPL package power limit driver
+======================================  ===============================
+
+Thermal Daemon source path in |C|
+---------------------------------
+
+======================================================  ===================================
+*vendor/intel/external/project-celadon/thermal_daemon*  Thermal daemon sources
```
Comment (klynnrif) [inline source/tutorials/thermal-daemon/thermal-daemon.rst]: Capitalize "daemon" / Thermal Daemon sources

File: source/getting_started/submit-patch.rst [hunk 11]
```diff
@@ -0,0 +1,86 @@
+.. _submit-patch:
+
+Submit your patches
+===================
+
+|C| uses `GitHub Pull-Requests <https://help.github.com/articles/about-pull-requests/>`_ (PR) to submit and review code submissions. You are required to be aware of the software repository and branch in |C| source tree where your code is contirbuting before submitting any changes.
```
[the remark this answers is not in the document]
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: done

File: source/getting_started/submit-patch.rst [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+|C| GitHub Manifest Structure
+-----------------------------
+
+|C| source code consists of multiple repositories ("**projects**"), the manifest file *.repo/manifests/default.xml* contains the information where ("**remotes**") to fetch those repositories. It includes the following files to instruct the ``repo`` command how to compose the |C| source tree:
```
[the remark this answers is not in the document]
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: thanks.

File: source/getting_started/submit-patch.rst [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+
+=============================  =============================================================
+include/aosp_vanilla.xml       This is vanilla AOSP code base on refs/tags/android-9.0.0_r18
```
[the remark this answers is not in the document]
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: done

File: source/getting_started/submit-patch.rst [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+include/remove-android_ia.xml  The repositories to be removed from the AOSP code
+include/bsp-celadon.xml        The BSP source code customized for |C|
+=============================  =============================================================
+
+Use the following command to get the *remote* where you are trying to upload code:
```
[the remark this answers is not in the document]
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: done

File: source/getting_started/submit-patch.rst [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+
+.. code-block:: bash
+
+   $ git remote -v
+
+The following table lists the *remote* aliases supported by |C|:
+
+=========  ========================================
+github     Good to go with github PR push process
```
[the remark this answers is not in the document]
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: thanks, but we need to keep the word `github` intact as it's the exact name in that include file.

File: source/getting_started/submit-patch.rst [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+aosp       Create a google_diff patch & in-parallel
```
[the remark this answers is not in the document]
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: I'm afraid not as they are the exact `remote aliases` defined in the manifest file.

File: source/getting_started/submit-patch.rst (hunk 1 above)
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: perfect!

File: source/getting_started/submit-patch.rst (hunk 2 above)
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: done

File: source/getting_started/submit-patch.rst (hunk 3 above)
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: you are professional!

File: source/getting_started/submit-patch.rst (hunk 4 above)
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: done

File: source/getting_started/submit-patch.rst (hunk 5 above)
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: done

File: source/getting_started/submit-patch.rst (hunk 6 above)
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: done

File: source/getting_started/submit-patch.rst (hunk 7 above)
Comment (ttzeng) [reply source/getting_started/submit-patch.rst]: thanks

File: source/tutorials/thermal-daemon/thermal-daemon.rst (hunk 8 above)
Comment (ttzeng) [reply source/tutorials/thermal-daemon/thermal-daemon.rst]: thanks

File: source/tutorials/thermal-daemon/thermal-daemon.rst (hunk 10 above)
Comment (ttzeng) [reply source/tutorials/thermal-daemon/thermal-daemon.rst]: done

File: source/tutorials/thermal-daemon/thermal-daemon.rst (hunk 9 above)
Comment (ttzeng) [reply source/tutorials/thermal-daemon/thermal-daemon.rst]: done

**Pull request merged**